component/quota-limitrange.jsonnet

```diff
@@ -28,7 +30,11 @@ local quotaSpec(rn, rq) =
     };
   spec {
     hard: std.foldl(function(x, k) x {
-      [k]: "{{ request.object.metadata.annotations.\"%s/%s.%s\" || '%s' }}" % [ 'resourcequota.appuio.io', rn, std.strReplace(k, '/', '_'), x[k] ],
+      [k]:
+        if std.length(std.findSubstr('storageclass.storage.k8s.io', k)) > 0 then
+          "{{ parse_json(request.object.metadata.annotations.\"%s/storageclasses\" || '{}').\"%s\" || '%s' }}" % [ quotaAnnotationPrefix, k, x[k] ]
```

Comment:
`parse_json()` is a custom JMESPath function provided by Kyverno, cf. https://kyverno.io/docs/writing-policies/variables/#general

Reply:
I think that one is fairly new. I do remember trying to do something like this and failing